### DISCUSSION TITLE
Changing the state of the PD on packet failure

### DISCUSSION THREAD

**Discussion by Master8oy** [goToMain/libosdp#186] [General]
I'm testing this library and I discovered one thing. When the PD does not send back a response, e.g. due to a failure or packet loss, the AC immediately marks the device as OFFLINE and starts communicating with the device as it did when it was first started. According to the standard, the AC should keep this PD in ONLINE state for 8 seconds and only then switch to OFFLINE state if it does not respond. Does this mean that this is not implemented in the library or can it be set somewhere?

**Comment by sidcha** [comment]
You are right, the library does not support this now. So far it didn't cause issues for me so didn't look into it any further.

We can add that if there is a pressing need/use-case for it.

**Comment by Master8oy** [comment]
Thanks for the info, I thought so :)
It would be good to at least have the option to activate this function, for example, in the configuration file. Although it does not cause major problems, it would be good to have this function in the library just because of the OSDP specification. So would it be possible to finish this? Thank you.

**Reply by sidcha** [reply]
Yes, makes sense. Also, this is required to fix another issue (#185) so I will look into this soon-ish.

**Comment by sidcha** [comment]
This issue been addressed in 182f049e0791b73a886b494a77f3b6551da18610
